crates/next-api/src/module_graph.rs

```diff
@@ -444,9 +526,12 @@ async fn get_module_graph_for_endpoint(
         )
         .to_resolved()
         .await?;
-        visited_modules.extend(graph.await?.graph.node_weights().map(|n| n.module));
+        visited_modules.extend(graph.await?.iter_nodes().map(|n| n.module));
         graphs.push(graph);
     }
+
+    // The previous iterations above (might) have added the entry node, but not actually visited it.
+    visited_modules.remove(&entry);
```

Comment:
How can that happen?

Reply:
This is actually intentional:
In the dev case, the graphs for each layout segment contain the rsc_entry (to be able to easily traverse them all with the same entry), which is added manually (and doesn't actually has its references visited).
And the visited_modules here are simply sourced from the nodes in the graph.

So to actually make this very last traversal do something, it has to remove the entry again.